### PR TITLE
Allow configuration of tolerations and nodeSelector for K8s hub agents

### LINKED ISSUE #942: Feature: allow configuration of nodeSelector and tolerations for spawned agents
## 🚀 Feature Proposal

Allow the user/admin to specify the Kubernetes nodeSelector and/or tolerations which are attached to spawned agents.

## Motivation

Split run location of hub, which is a permanent service, from the agents, which are preemptible.
This would allow running hub on a nodepool with permanent nodes and the agents on an on-demand nodepool.

## Example

Example config:
```
env:
  - name: ZALENIUM_KUBERNETES_TOLERATIONS
    value: "---\n- effect: \"NoExecute\"\n  key: \"no-cluster-autoscaling\"\n  operator: \"Equal\"\n  value: \"false\"\n- effect: \"NoExecute\"\n  key: \"cluster-autoscaling\"\n  operator: \"Equal\"\n  value: \"true\""
  - name: ZALENIUM_KUBERNETES_NODE_SELECTOR
    value: "beta.kubernetes.io/arch=linux", kops.k8s.io/instancegroup=no-cluster-autoscaling"
```

would translate in PodSpec to
```
Node-Selectors:
  - beta.kubernetes.io/arch=linux
  - kops.k8s.io/instancegroup=no-cluster-autoscaling
Tolerations:
  - no-cluster-autoscaling=false:NoExecute
  - cluster-autoscaling=true:NoExecute
```

### PR DESCRIPTION
**Pull request by schlippi**
### Description
Lookup env variables `ZALENIUM_KUBERNETES_TOLERATIONS` and `ZALENIUM_KUBERNETES_NODE_SELECTOR` to apply them to the hub agents spawned in Kubernetes.
The current behavior is changed in that first env var lookup is done, otherwise old behavior is executed.
This PR closes #942.

### Motivation and Context
This change allows the user to have hub and agents running on different nodepools what is currently not possible.

### How Has This Been Tested?
As this is a little change, I only wrote a couple of unit tests.

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by schlippi** [comment]
Some hints where to update the documentation for the env variables would be nice.

File: src/main/java/de/zalando/ep/zalenium/container/kubernetes/KubernetesContainerClient.java [hunk 1]
```diff
@@ -156,16 +159,26 @@ private void discoverHostAliases() {
     }
 
     private void discoverNodeSelector() {
-        final Map<String, String> configuredNodeSelector = zaleniumPod.getSpec().getNodeSelector();
-        if (configuredNodeSelector != null && !configuredNodeSelector.isEmpty()) {
-            nodeSelector = configuredNodeSelector;
+        final Map<String,String> nodeSelectorFromEnv = environment.getMapEnvVariable(ZALENIUM_KUBERNETES_NODE_SELECTOR, new HashMap<>());
+        if (nodeSelectorFromEnv != null && !nodeSelectorFromEnv.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (schlippi) [reply src/main/java/de/zalando/ep/zalenium/container/kubernetes/KubernetesContainerClient.java]: As the returned variable is either a map, or like below a list, it is not possible to use the `Strings.isNullOrEmpty()` method you suggested. 
What I could think of is checking for the existence of the env var and moving the call to fetch and transform the env variable values into the `if` branch. You ok with that?
```suggestion
if(!Strings.isNullOrEmpty(environment.getEnvVariable(ZALENIUM_KUBERNETES_NODE_SELECTOR))) {
  nodeSelector = environment.getMapEnvVariable(ZALENIUM_KUBERNETES_NODE_SELECTOR, new HashMap<>());
}
```

[the remark this answers is not in the document]
Comment (diemol) [reply src/main/java/de/zalando/ep/zalenium/container/kubernetes/KubernetesContainerClient.java]: Your are completely right, I missed that, thanks!

**Comment by andre1987** [comment]
@diemol when this feature will be released?

**Comment by diemol** [comment]
@andre1987 we just released

**Pull request merged**